addOns/oast/src/main/java/org/zaproxy/addon/oast/services/interactsh/InteractshService.java

```diff
@@ -271,17 +272,19 @@ public synchronized void deregister() {
             HttpMessage deregisterMsg = new HttpMessage(reqHeader, reqBody);
             httpSender.sendAndReceive(deregisterMsg);
             if (deregisterMsg.getResponseHeader().getStatusCode() != 200) {
+
                 LOGGER.warn(
                         "Error during interactsh deregister, due to bad HTTP code {}. Content: {}",
                         deregisterMsg.getResponseHeader().getStatusCode(),
                         deregisterMsg.getResponseBody());
-                return;
             }
             LOGGER.debug("Deregistered correlationId: {}", correlationId);
-            isRegistered = false;
-            fireOastStateChanged(new OastState(getName(), false, null));
+
         } catch (Exception e) {
             LOGGER.error("Error during interactsh deregister: {}", e.getMessage(), e);
+        } finally {
+            isRegistered = false;
+            fireOastStateChanged(new OastState(getName(), false, null));
```

Comment:
The InteractshOptionsPanelTab instance should register as an OastStateChangedListener and clear the payloads table when the service is deregistered.

Reply:
This should be already happening?

Reply:
I don't think we are doing that currently.

Reply:
Right, was asking if it was something that should have been done before these changes. I'm fine if done in this PR.